### PR TITLE
Catch panic in built-in connectors

### LINKED ISSUE #220: Plugin: Catch panics in built-in connectors
### Feature description

Built-in connector plugins run in the same process as Conduit, a badly implemented plugin could potentially bring down the whole process with a panic. We need to make sure built-in plugins run in some kind of a sandbox where we catch all panics and transform them into an error instead.

Standalone plugins should not be affected by this since panics should be handled by the framework, but we need to test this out to be sure.

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

This PR makes sure explicit calls to builtin plugins can't crash Conduit. If the plugin spawns a goroutine that panics it can still bring down Conduit, there's no way to catch those panics (if there is, let me know!).

First use of generics in Conduit! 😁 

Fixes #220

### Quick checks:

- [X] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [X] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [X] I have written unit tests.
- [X] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/plugin/builtin/v1/source.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (s *sourcePluginAdapter) Start(ctx context.Context, p record.Position) erro
 	s.stream = newSourceRunStream(ctx)
 	go func() {
 		s.logger.Trace(ctx).Msg("calling Run")
-		err := s.impl.Run(s.withLogger(ctx), s.stream)
+		err := runSandboxNoResp(s.impl.Run, s.withLogger(ctx), cpluginv1.SourceRunStream(s.stream))
```
Comment (dylanlott) [inline pkg/plugin/builtin/v1/source.go:99]: Do we need to sandbox the `SourceRunStream` too, like the comment for `DestinationRunStream` notes?

**Review by dylanlott — APPROVED** [review]
I was wondering when we'd see generics first used. Looks good to me. ✨

File: pkg/plugin/builtin/v1/source.go (hunk 1 above)
Comment (lovromazgon) [reply pkg/plugin/builtin/v1/source.go:98]: Which comment do you mean? I don't think we need to sandbox anything else, actually, I don't think we _can_ sandbox anything else. Conduit calls 5 functions in the plugin and those functions are all sandboxed now.

Comment (dylanlott) [reply pkg/plugin/builtin/v1/source.go:98]: I thought there was a TODO comment about sandboxing the SourceRunStream, but I don't see it anymore so disregard 👍

**Pull request merged**